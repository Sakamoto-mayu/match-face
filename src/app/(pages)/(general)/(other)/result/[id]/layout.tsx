export const metadata = {
  title: "回答確認",
  description: "一般ユーザー回答確認ページ",
};

export default function ResultLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
