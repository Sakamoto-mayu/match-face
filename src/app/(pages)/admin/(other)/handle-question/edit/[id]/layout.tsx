export const metadata = {
    title: "質問・解答例編集",
    description: "管理者質問・解答例編集ページ",
  };
  
  export default function editLayout({
    children,
  }: {
    children: React.ReactNode;
  }) {
    return <>{children}</>;
  }
  