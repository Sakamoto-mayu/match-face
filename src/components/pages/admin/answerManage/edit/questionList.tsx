import type { Question } from "@/const/projectTable";

interface QuestionListProps {
  questions: Question[];
}

const QuestionList = (props: QuestionListProps) => {
  const { questions } = props;
  return (
    <>
      {questions.map((data) => {
        if (data.answer_example) {
          return (
            <div key={data.question_id}>
              <h2>Q{data.question_id}.</h2>
              <textarea
                name="question"
                id="question"
                value={data.question}
                className="border-deep-gray"
              ></textarea>
              <label htmlFor="answer_ex">回答例</label>
              <textarea
                name="answer_ex"
                id="answer_ex"
                value={data.answer_example}
                className="border-deep-gray"
              ></textarea>
            </div>
          );
        } else if (data.answer) {
          return (
            <div key={data.question_id}>
              <h2>Q{data.question_id}.</h2>
            </div>
          );
        }
      })}
    </>
  );
};

export default QuestionList;
