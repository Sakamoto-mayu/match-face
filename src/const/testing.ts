export const project = [
  {
    id: "1",
    name: "案件のタイトルが入ります",
    date: new Date("2023/07/01"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "2",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-02"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
  {
    id: "3",
    name: "案件のタイトルが入ります",
    date: new Date("2023-07-03"),
    project_detail:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
  },
];

export const question = [
  {
    id: "1",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "2",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "3",
    project_id: "1",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "4",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "5",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "6",
    project_id: "2",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "7",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
  {
    id: "8",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "select",
    choices: ["選択肢A", "選択肢B", "選択肢C", "選択肢D"],
  },
  {
    id: "9",
    project_id: "3",
    name: "質問のタイトルが入ります",
    content:
      "テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト",
    type: "writing",
  },
];
