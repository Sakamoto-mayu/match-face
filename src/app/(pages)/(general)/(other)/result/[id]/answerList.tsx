import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "@/components/pages/general/result/ModelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}:{question.name}
          </h3>
          <p id={`content-${index + 1}`} className="mt-2 mb-4">
            {question.content}
          </p>
          {question.type === "writing" ? (
            <div
              id={`answer-${index + 1}`}
              className="border border-gray p-3 mt-2 rounded-md shadow-md"
            >
              <h3 className="mb-2 text-xl">あなたの回答</h3>
              {currentAnswer[index].content}
            </div>
          ) : (
            <>
              <ul>
                {question.choices?.map((choice, index) => (
                  <li key={index}>・{choice}</li>
                ))}
              </ul>
              <div
                id={`answer-${index + 1}`}
                className="border border-gray p-3 mt-5 rounded-md shadow-md"
              >
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
