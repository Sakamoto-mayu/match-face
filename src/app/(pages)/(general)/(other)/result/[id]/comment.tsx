import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import { User } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: number;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );
  const commentUser = User.filter(
    (user) => user.id === currentComment[0].admin_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">{commentUser[0].name}からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={commentUser[0].image_url}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
            priority
          />
        </div>
        <div id="comment" className="w-5/6">
          {currentComment[0].content}
        </div>
      </div>
    </>
  );
};
