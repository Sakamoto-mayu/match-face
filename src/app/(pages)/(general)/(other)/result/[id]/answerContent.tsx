"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";
import { Answer } from "@/const/result";
import { useRouter } from "next/navigation";

type Props = {
  user_id: number;
  project_id: number;
};

export const AnswerContent = ({ user_id, project_id }: Props) => {
  const router = useRouter();
  const currentProject = project.filter(
    (project) => project.id === Number(project_id)
  );
  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === Number(project_id)
  );

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        {/*現状のデータでは4以降の解答データを呼び出そうとするとエラーになります */}
        <p>回答日時:{currentAnswer[0].answer_date.toLocaleDateString()}</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">案件名：{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList project_id={project_id} />
      <CommentContent user_id={user_id} project_id={project_id} />
      <div className="flex justify-around my-20">
        <OrangeButton
          className="border rounded-none"
          label="質問一覧へ"
          onClick={() => router.push("/questions")}
        />
        <OrangeButton
          className="border rounded-none"
          label="回答履歴一覧へ"
          onClick={() => router.push("/histories")}
        />
      </div>
    </div>
  );
};
