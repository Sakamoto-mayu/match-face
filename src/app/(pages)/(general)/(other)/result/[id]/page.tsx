import { AnswerContent } from "./answerContent";

const ResultPage = ({ params }: { params: { id: number } }) => {
  const user_id = 1;

  return (
    <div className="flex flex-col items-center">
      <AnswerContent user_id={user_id} project_id={params.id} />
    </div>
  );
};

export default ResultPage;
