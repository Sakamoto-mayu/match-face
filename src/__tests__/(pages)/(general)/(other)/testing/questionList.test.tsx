import { QuestionList } from "@/app/(pages)/(general)/(other)/testing/[id]/questionList";
import { render, screen } from "@testing-library/react";
import { question } from "@/const/testing";
import "@testing-library/jest-dom";

const testData = {
  id: 1,
};

const currentQuestion = question.filter(
  (question) => question.project_id === testData.id
);

describe("QuestionListコンポーネントのテスト", () => {
  test("質問タイトルと詳細内容が表示される", () => {
    render(<QuestionList id={testData.id} />);
    // 質問タイトルと質問詳細内容がデータと一致する
    currentQuestion.map((question, index) => {
      const heading = screen.getByRole("heading", {
        level: 2,
        name: `Q${index + 1}:${question.name}`,
      });
      const content = document.querySelector(`#question-content-${index}`);

      expect(heading).toBeInTheDocument();
      expect(content).toHaveTextContent(`${question.content}`);
    });
  });
  test("type=writingの場合、解答欄が表示される", () => {
    render(<QuestionList id={testData.id} />);
    currentQuestion.map((question, index) => {
      if (question.type === "writing") {
        const title = document.querySelector(`#answer-title-${index}`);
        const textarea = document.querySelector(`#${question.name}`);
        expect(title).toBeInTheDocument();
        expect(textarea).toBeInTheDocument();
      }
    });
  });
  test("type=selectの場合、選択肢が表示される", () => {
    render(<QuestionList id={testData.id} />);
    currentQuestion.map((question, index) => {
      if (question.type === "select") {
        question.choices?.map((choice, index) => {
          const input = document.querySelector(`#choice-${index}`);
          const label = document.querySelector(`#label-${index}`);
          expect(input).toBeInTheDocument();
          expect(label).toBeInTheDocument();
        });
      }
    });
  });
  test("スナップショットテスト", () => {
    const { container } = render(<QuestionList id={testData.id} />);
    expect(container).toMatchSnapshot();
  });
});
