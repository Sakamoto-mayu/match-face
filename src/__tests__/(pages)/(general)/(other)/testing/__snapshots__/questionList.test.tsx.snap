// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionListコンポーネントのテスト スナップショットテスト 1`] = `
<div>
  <div
    class="py-5"
  >
    <h2>
      Q
      1
      :
      質問のタイトルが入ります
    </h2>
    <p
      class="mt-2 mb-4"
      id="question-content-0"
    >
      テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
    </p>
    <label
      for="質問のタイトルが入ります"
    >
      <p
        id="answer-title-0"
      >
        解答記入欄
      </p>
    </label>
    <textarea
      class="border border-black mt-1 w-full"
      cols="50"
      id="質問のタイトルが入ります"
      name="質問のタイトルが入ります"
      rows="6"
    />
  </div>
  <div
    class="py-5"
  >
    <h2>
      Q
      2
      :
      質問のタイトルが入ります
    </h2>
    <p
      class="mt-2 mb-4"
      id="question-content-1"
    >
      テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
    </p>
    <div>
      <input
        id="choice-0"
        name="質問のタイトルが入ります"
        type="radio"
        value="選択肢A"
      />
      <label
        for="choice0"
        id="label-0"
      >
        選択肢A
      </label>
    </div>
    <div>
      <input
        id="choice-1"
        name="質問のタイトルが入ります"
        type="radio"
        value="選択肢B"
      />
      <label
        for="choice1"
        id="label-1"
      >
        選択肢B
      </label>
    </div>
    <div>
      <input
        id="choice-2"
        name="質問のタイトルが入ります"
        type="radio"
        value="選択肢C"
      />
      <label
        for="choice2"
        id="label-2"
      >
        選択肢C
      </label>
    </div>
    <div>
      <input
        id="choice-3"
        name="質問のタイトルが入ります"
        type="radio"
        value="選択肢D"
      />
      <label
        for="choice3"
        id="label-3"
      >
        選択肢D
      </label>
    </div>
  </div>
  <div
    class="py-5"
  >
    <h2>
      Q
      3
      :
      質問のタイトルが入ります
    </h2>
    <p
      class="mt-2 mb-4"
      id="question-content-2"
    >
      テキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキストテキスト
    </p>
    <label
      for="質問のタイトルが入ります"
    >
      <p
        id="answer-title-2"
      >
        解答記入欄
      </p>
    </label>
    <textarea
      class="border border-black mt-1 w-full"
      cols="50"
      id="質問のタイトルが入ります"
      name="質問のタイトルが入ります"
      rows="6"
    />
  </div>
</div>
`;
